srcts/src/shiny/shinyapp.ts

```diff
@@ -1112,8 +1112,8 @@ class ShinyApp {
           // loop through all existing tabs, find the one with highest id
           // (since this is based on a numeric counter), and increment
 
-          $tabset.find("a[data-toggle='tab']").each(function () {
-            const $tab = $(this);
+          $tabset.find("> li").each(function () {
+            const $tab = $(this).find("> a[data-toggle='tab']");
```

Comment:
This reverts the logic back to the way it was in the 1.6.0 release (https://github.com/rstudio/shiny/blob/b57cb6c8e1ed2e8d75ea0ba47854b15e03bc1287/srcjs/shinyapp.js#L876-L885)

I changed originally this logic to accommodate the fact that, for a short period, we were working around a BS4 regression that dropped support for dropdown items wrapped in `<li>`. That regression has since [been fixed](https://github.com/twbs/bootstrap/pull/33649) and is now included with `bs_theme(version = 4)`, so this change in logic is no longer needed